Jellyfin.Api/Controllers/NotificationsController.cs

```diff
@@ -0,0 +1,137 @@
+        /// <param name="notificationManager">The notification manager.</param>
+        /// <param name="userManager">The user manager.</param>
+        public NotificationsController(INotificationManager notificationManager, IUserManager userManager)
+        {
+            _notificationManager = notificationManager;
+            _userManager = userManager;
+        }
+
+        /// <summary>
+        /// Endpoint for getting a user's notifications.
+        /// </summary>
+        /// <param name="userId">The user's ID.</param>
+        /// <param name="isRead">An optional filter by IsRead.</param>
+        /// <param name="startIndex">The optional index to start at. All notifications with a lower index will be dropped from the results.</param>
+        /// <param name="limit">An optional limit on the number of notifications returned.</param>
+        /// <returns>A read-only list of all of the user's notifications.</returns>
+        [HttpGet("{UserID}")]
+        public IReadOnlyList<NotificationDto> GetNotifications(
+            [FromRoute] string userId,
+            [FromQuery] bool? isRead,
+            [FromQuery] int? startIndex,
+            [FromQuery] int? limit)
+        {
+            return new List<NotificationDto>();
+        }
+
+        /// <summary>
+        /// Endpoint for getting a user's notification summary.
+        /// </summary>
+        /// <param name="userId">The user's ID.</param>
+        /// <returns>Notifications summary for the user.</returns>
+        [HttpGet("{UserID}/Summary")]
+        public NotificationsSummaryDto GetNotificationsSummary(
+            [FromRoute] string userId)
+        {
+            return new NotificationsSummaryDto();
+        }
+
+        /// <summary>
+        /// Endpoint for getting notification types.
+        /// </summary>
+        /// <returns>All notification types.</returns>
+        [HttpGet("Types")]
+        public IEnumerable<NotificationTypeInfo> GetNotificationTypes()
+        {
+            return _notificationManager.GetNotificationTypes();
+        }
+
+        /// <summary>
+        /// Endpoint for getting notification services.
+        /// </summary>
+        /// <returns>All notification services.</returns>
+        [HttpGet("Services")]
+        public IEnumerable<NameIdPair> GetNotificationServices()
+        {
+            return _notificationManager.GetNotificationServices();
+        }
+
+        /// <summary>
+        /// Endpoint to send a notification to all admins.
+        /// </summary>
+        /// <param name="name">The name of the notification.</param>
+        /// <param name="description">The description of the notification.</param>
+        /// <param name="url">The URL of the notification.</param>
+        /// <param name="level">The level of the notification.</param>
+        [HttpPost("Admin")]
+        public void CreateAdminNotification(
+            [FromForm] string name,
+            [FromForm] string description,
```

Comment:
I think `name` and `description` need the `[Required]` attribute and then some check of the `ModelState` to make sure they have been provided. I'm not sure what the existing API does if these aren't provided, but they are marked with `IsRequired = true` there.

Maybe someone else in @jellyfin/backend knows how validation works in the existing API.

Reply:
The ASP.NET framework will actually automatically mark any parameter that isn't nullable in the handler method as required. Sending a POST request to `localhost:8096/Notifications/Admin` after converting the fields to query parameters, will return the following error message:
```
{
    "type": "https://tools.ietf.org/html/rfc7231#section-6.5.1",
    "title": "One or more validation errors occurred.",
    "status": 400,
    "traceId": "|ba724dde-4825094e2d915a31.",
    "errors": {
        "name": [
            "The name field is required."
        ],
        "description": [
            "The description field is required."
        ]
    }
}
```